shared/vaccineDataStore.ts

```diff
@@ -0,0 +1,54 @@
+import { NzTimeIso, VaccineData } from "./types/VaccineDataTypes";
+import { promises as fs, existsSync as fileExists } from "fs";
+import { DateTime } from "luxon";
+import { CONSTANTS } from "./constants";
+
+export const createOrUpdateVaccineDataForDate = async (
+  time: NzTimeIso,
+  vaccineData: VaccineData
+) =>
+  await fs.writeFile(
+    generateVaccineDataFilePath(time),
+    JSON.stringify(vaccineData, null, 2)
+  );
+
+export const getVaccineDataForDate = async (
+  time: NzTimeIso
+): Promise<VaccineData | null> => {
+  const filePath = generateVaccineDataFilePath(time);
+  if (fileExists(filePath)) {
+    const data = await fs.readFile(filePath, "utf8");
+    return JSON.parse(data);
+  }
+  return null;
+};
+
+export const getAllVaccineData = async (
+  // by default return all data
+  maxItems: number = 999
+): Promise<VaccineData[]> => {
+  const files = await fs.readdir(CONSTANTS.vaccineDataFolder);
+
+  const vaccineData: VaccineData[] = [];
+  const filesToParse = files.sort().reverse().slice(0, maxItems);
+
+  for (const f of filesToParse) {
+    const filePath = `${CONSTANTS.vaccineDataFolder}/${f}`;
+    const data = await fs.readFile(filePath, "utf8");
+    vaccineData.push(JSON.parse(data));
+  }
+
+  return vaccineData;
```

Comment:
Could use promise.all? Run in parallel?

```suggestion
  const files = await fs.readdir(CONSTANTS.vaccineDataFolder);
  const filesToParse = files.sort().reverse().slice(0, maxItems);

  const vaccineData = filesToParse.map(async (f) => {
    const filePath = `${CONSTANTS.vaccineDataFolder}/${f}`;
    const data = await fs.readFile(filePath, "utf8");
    return JSON.parse(data);
  });

  return Promise.all(vaccineData);
```

Reply:
ooo great idea